### ISSUE TITLE
Build single executable with PyInstaller

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/clcache-feedstock#2]
It appears one can build a binary of clcache with PyInstaller as noted in the [docs]( https://github.com/frerich/clcache/#installation-via-exe ). This would remove the Python dependency and allow this to work with VS 2008 builds. Could be useful for caching build artifacts as noted [here]( https://github.com/conda-forge/llvmdev-feedstock/pull/18#issuecomment-288891833 ). Thoughts on this?

**Comment by jakirkham** [comment]
cc @gqmelo

**Comment by gqmelo** [comment]
Thanks, but now I realize we already solved this problem at work. We [created a chocolatey package](https://github.com/frerich/clcache/pull/267) and it can be installed using ```choco install clcache```.
We can use chocolatey on AppVeyor,  right?

**Comment by tadeu** [comment]
Yes, what do you propose, a recipe installing it via `choco` inside the `Scripts` dir, and packaging it? Seems good.

**Comment by gqmelo** [comment]
Not really, I was just proposing to use the choco package on Appveyor builds instead of the conda package and close this issue. But if it is easy to do what you said, even better. Can I make choco install it in $PREFIX?

**Comment by tadeu** [comment]
Meh, apparently only using the licensed version of `choco` :/

Installing the PyInstaller version of `clcache` via `choco` seems better anyway, because it puts only `clcache.exe` in the PATH, all DLL's (python, etc.) stay in a subdirectory.

The PyInstaller version alone comes with a lot of binaries in the same PATH directory (`python34.dll`, `pyexpat.pyd`, `MSVCR100.dll`), etc., which would polllute the conda env. There is a flag `--onefile` that can make it generate a single-file executable, but the generated executable is very slow (every call have to unzip a lot of files and put them on the filesystem).

**Comment by gqmelo** [comment]
> Meh, apparently only using the licensed version of choco :/

Yes, I've seen that. But I think there is an alternative. The recipe can use pyinstaller to create the files on a dir which will not be on PATH. Then it could a create a shim in ```%PREFIX%\Scripts```. There is a powershell function provided by chocolatey for that:

https://chocolatey.org/docs/helpers-install-bin-file

**Comment by jakirkham** [comment]
> The PyInstaller version alone comes with a lot of binaries in the same PATH directory (`python34.dll`, `pyexpat.pyd`, `MSVCR100.dll`), etc., which would polllute the conda env. There is a flag `--onefile` that can make it generate a single-file executable, but the generated executable is very slow (every call have to unzip a lot of files and put them on the filesystem).

Maybe something else to consider would be using [Nuitka]( http://nuitka.net/pages/overview.html ) to build a `clcache` binary executable. We already have a [`nuitka`]( https://github.com/conda-forge/nuitka-feedstock ) package.

Edit: Also upstream issue ( https://github.com/frerich/clcache/issues/232 ) seems relevant.

**Comment by jakirkham** [comment]
Other methods for generating a Windows executable include [cx_Freeze]( https://cx-freeze.readthedocs.io/ ) and [py2exe]( http://www.py2exe.org/ ).
